### DISCUSSION TITLE
WinUI Community Call March 2024 Q&A Follow-up

### DISCUSSION THREAD

**Discussion by duncanmacmichael** [microsoft/microsoft-ui-xaml#9572] [General]
Hi everyone, since we received a lot of questions on the March 2024 call's live comments that we didn't have time to answer, we'd like to follow up here and make sure we address all the ones we couldn't get to during the call. We appreciate your patience and we want to thank you as always for your great feedback and questions. We'd also like to take this opportunity to sum up what we did answer on the call, as well as call out upcoming events!

## Summary of questions from the live comments that were answered during the call

> NOTE: Questions are presented verbatim from their original post.

### Q: is the new WinUI 3 gallery app updated with this?

**A:** Yes, the gallery app was updated this week \[of the call\] with the new controls.

### Q: Is there any plan to replace the MapControl by a native control instead of the WebView-based one? Offline access is a must for our use cases.

**A:** No plans.

### Q: How is the SelectorBar any different from NavigationView top mode if you the content isn't bound to the items?

**A:** NavigationView was heavy-weight. It's great if you wanted to swap dynamically between left/top, but SelectorBar is lighter and more performant.

### Q: ​​@Scott Jones can we fix the breakpoint though so Visual Studio instead breaks further up the callstack in my measureoverride code?

**A:** Good feedback, looking to optimize. It's continuing to evolve. Getting source available was the first step, continuing to evolve the experience here in this space. We're planning to work more in 1.6 on Intellisense as well, since we want the whole platform to be easy to use.

### Q: ​​Any chance to get NativeAOT before 1.6? ​​What is the planned schedule for the 1.6?

**A:** That would have to be a 1.5 servicing release, so very unlikely. 1.6 is coming in Fall 2024.

### Q: ​​When will you truly open source WinUI, allowing us to build it ourself and contribute to it?

**A:** This was answered in the pre-call questions on the community call discussion post, re-posted here: We've been careful to use the term "source available", which, with public debug symbols, enables debugging the Windows App SDK binaries. Full "open source", with the expectation of buildability, testability, accepting PRs, and so on, is still work in progress. No updates on that.

### Q: ​​Q: Can you make the latest WinUI3 code the main branch? You can't search the source code when it's in a branch

**A:** We're aware that the current publishing of the branch is hard to search and are hoping to address this.

### Q: ​​Any plans to make source available other parts of WinAppSDK (e.g. Storage API)?

**A:** We'd like to make things available but there are lots of moving parts, some integrated into the Windows build process. So it's a journey and a goal. The intent is to continue to do this. Some things are still called from within the OS itself vs. the WinAppSDK, and those aren't planned to open source.

### Q: ​​are there any official plans for supporting WinAppSdk outside visual studio?

**A:** Things should be doable in the command line with msbuild. We don't have other plans beyond that, like cmake. There's not any real dependency on Visual Studio. 

In response to a further question about having to use msbuild, we also responded that there is no reason we shouldn't be able to support dotnet builds.

### Q: ​​Q: What is the point of experimental releases? You shipped 1.5 even though there was a critical issue reported in all the experimental releases from the first day they shipped. ... and you shipped 1.5 without addressing it until 1.5.1

**A:** This is something we're aware of - multiple facets come into play. Last summer, we brought issues on GitHub to our internal bug database to triage a single set of bugs, and we have triaged the whole GitHub issue list. We're addressing them to have higher visibility. Before, we had too long before we'd see a bug before we could respond. So we are tightening that loop. in 1.6, we're putting resources into infrastructure to make it easier to do releases, so we can focus on getting feedback and acting on fixes.

### Q: ​​Q. any update on unsealing all the default UI controls?

**A:** The biggest slowdown is ensuring binary compatibility. We've been working towards that, with plans for ItemsGrid in 1.6. If that works, we will do more in 1.7.

### Q: ​​How about support for CaptureElement to make webcam integration easier?

**A:** No plans for CaptureElement - MediaPlayerElement should be able to do this. There's an examlpe in the WinUI 3 Gallery app.

## Questions asked in the live comments that were not answered during the call

> NOTE: If there is a question you asked but don't see here, it might be something we're unclear about or don't have a comment for at this time, it might have been a duplicate of another similar question, or it could be just something we missed. Feel free to open a new discussion or raise again during our next community call Q&A.

### Q: ​​There is reported issues about localization regarding command bar icon labels in Windows Insider Builds. has this been addressed for the 1.5 release or is the issue still being tracked?

**A:** The labels are coming from File Explorer and are being tracked by the File Explorer team.

### Q: "​​Will the ink controls be native using the original WinRT APIs, or will it be a WebView implementation like the new MapControl?

**A:** The goal is a native control.

### Q: ​​Can you elaborate on x:Bind improvements? Sounds promising

**A:** This is to address a set of x:Bind bug reports on GitHub.

### Q: ​​Were there any improvements in the releasing of memory resources on the closing of windows?

**A:** There has been some work here and there is more to do. It isn't currently clear what progress might be made in 1.6.

### Q: ​​Any plans to bring Storage API features to File IO (e.g. thumbnails)?

**A:** The Windows.Storage APIs are available for use (with some limitations, like not being able to use "ms-appx:///" Uris outside of UWP), and any other .NET or native Windows file API is also available for use. There aren't currently any plans to create WinAppSDK-specific Storage APIs.

### Q: ​​What about hot reloading ?

**A:** Hot reloading is supported today. If you are hitting scenarios where it doesn't work, please report them on GitHub.

### Q: ​​what happened to input validation apis that were in the 1.5 experimental releases but removed from 1.5 stable?

**A:** The Input Validation APIs which have been in experimental builds are old prototype code. Much work is required to complete this feature. This feature is not part of our 1.6 plans and has been removed from future experimental builds.

### Q: ​​Any plans to be able to run WinUI 3 without XAML?Currently you have to implement IXamlMetadataProvider eg.public partial class App : Application, IXamlMetadataProvider

**A:** No, there aren't any plans for that.

### //Build

Microsoft //Build is coming up on May 21-23 in Seattle, with news about AI, Copilots, and more. Stay tuned for WinUI-related updates from the conference!

### Next WinUI Community Call

The next WinUI community call will take place on June 5th, 2024. As we approach the call date, we'll post a discussion to gather questions ahead of time and share the call agenda. We look forward to seeing you then!

### More content to come!

We heard the feedback from the post-call survey that you'd like to see content more frequently than each quarter, and we're exploring options for that as we continue our efforts in improving engagement with our community. Keep an eye out for announcements soon!

**Comment by nkasco** [comment]
Can a Hello World solution that runs a basic WinUI 3 form within a PowerShell 7 host be provided? There are many sysadmins who create PowerShell based tools and desire a visually appealing, modern, UI Framework other than WinForm or WPF. There doesn't seem to be guidance available to identify the required dependencies that must be loaded.

**Reply by duncanmacmichael** [reply]
Great question! I'd suggest opening a feature request for this. We'll be more than happy to take a look at it, although I can't comment on a time frame right now. Thanks!

**Reply by nkasco** [reply]
Thanks for the response. I have submitted a feature request here: #9577
